src/artifacts.js

```diff
@@ -175,8 +217,46 @@ api.declare({
   // Construct details for different storage types
   var isPublic = /^public\//.test(name);
   var details  = {};
+  let uploadId;
   switch (storageType) {
+    case 'blob':
+      // Generate the details that we'll be using.
+      details.size = input.size;
+      details.sha256 = input.sha256;
+      if (input.parts) {
```

Comment:
If we `createArtifact` a second time with a different set of parts then it's an idempotency error if we return the same upload id for a different set of parts.

Why would we care to transform the parts when we can just store the original one and always work off it?  Maybe we can delete the parts list during `completeArtifact`, but I'd rather that we keep that info around for later examination.